duckdb-vortex/src/include/vortex_common.hpp

```diff
@@ -1,36 +1,33 @@
 #pragma once
+#define ENABLE_DUCKDB_FFI
 
 #include "duckdb.hpp"
+#include "duckdb/common/unique_ptr.hpp"
+
 #include "vortex.hpp"
 #include "vortex_error.hpp"
 
-#include <duckdb/common/unique_ptr.hpp>
+namespace vortex {
 
 struct DType {
-	explicit DType(vx_dtype *dtype): dtype(dtype) {}
+	explicit DType(vx_dtype *dtype) : dtype(dtype) {
+	}
 
-	static duckdb::unique_ptr<DType> FromDuckDBTable(
-		const std::vector<duckdb_logical_type> &column_types,
-		const std::vector<unsigned char> &column_nullable,
-		const std::vector<const char *> &column_names
-	) {
+	static duckdb::unique_ptr<DType> FromDuckDBTable(const std::vector<duckdb_logical_type> &column_types,
+	                                                 const std::vector<unsigned char> &column_nullable,
+	                                                 const std::vector<const char *> &column_names) {
 		D_ASSERT(column_names.size() == column_nullable.size());
 		D_ASSERT(column_names.size() == column_types.size());
 
 		vx_error *error = nullptr;
-		auto dtype = vx_duckdb_logical_type_to_dtype(
-			column_types.data(),
-			column_nullable.data(),
-			column_names.data(),
-			column_names.size(),
-			&error
-		);
-		HandleError(error);
+		auto dtype = Try([&](auto err) {
+			return vx_duckdb_logical_type_to_dtype(column_types.data(), column_nullable.data(), column_names.data(),
+			                                       column_names.size(), &error);
```

Comment:
Should this be `err`?

Reply:
How do you mean? The wrapping `Try` handles the error in case `vx_duckdb_logical_type_to_dtype` fails.

Reply:
```
template <typename Func>
auto Try(Func func) {
	vx_error *error = nullptr;
	auto result = func(&error);
	HandleError(error);
	return result;
}
```

Reply:
Oh now I see what you mean, I was still using the prev `error` variable. => Changed.